src/gpt_resolve/pdf_generator.py

```diff
@@ -3,6 +3,33 @@
 from pylatex import Document, Command
 from pylatex.utils import NoEscape
 import datetime
+import tempfile
+import subprocess
+
+def compile_solution(content: str) -> bool:
+    """
+    Compile a LaTeX document to check if the content has valid LaTeX syntax.
+
+    Args:
+        content (str): The LaTeX content to check.
+
+    Returns:
+        bool: True if compilation is successful, False otherwise.
+    """
+    with tempfile.TemporaryDirectory() as temp_dir:
+        tex_file = Path(temp_dir) / "temp_solution.tex"
+        doc = Document()
+        doc.append(NoEscape(content))
+        doc.generate_tex(str(tex_file))
+        try:
+            result = subprocess.run(
+                ["pdflatex", "-output-directory", temp_dir, tex_file],
+                capture_output=True, text=True
+            )
+            return result.returncode == 0
+        except subprocess.CalledProcessError as e:
+            print(f"Error compiling LaTeX content: {e}")
+            return False
```

Comment:
About this function, I have two suggestions:
- a more clear name like validate_latex_content since it returns a bool, with docstring like "Validate a LaTeX document by attempting to compile it, checking for syntax validity."
- subprocess does not seem necessary, as you can use `doc.generate_pdf(tex_file)`, right?